forge16/src/main/java/com/envyful/better/dex/rewards/forge/listener/DexRewardsListener.java

```diff
@@ -32,6 +32,21 @@ public void onPokemonCatch(PokedexEvent event) {
             return;
         }
 
+        if (event.pokemon == null) {
+            return;
+        }
+
+        if (this.mod.getConfig().getRequiresOriginalTrainerToReward()) {
+            if (event.pokemon.getOriginalTrainerUUID() == null) {
```

Comment:
Do we want to return if the OT UUID is null? Is this not null when it's from a wild pokemon?

Reply:
Okay, so we don't want to just return out of the event if it is null then?

Reply:
No? If the check is removed, an NPE is thrown when you're in battle with the Pokémon. When you catch it, the event is fired again and registered to the dex and mod

Reply:
I'm not suggesting you remove the check. The NPE fires because of the secondary check you add, so rather than returning out of the event you wrap your check with the null one